src/components/sidebar/index.js

```diff
@@ -81,6 +82,15 @@ const Sidebar = ({}) => {
     fetchStore()
   }, [])
 
+  const triggerHandler = () => {
+    const id = triggerHandler.id++
+    return {
+      open: currentlyOpen === id,
+      handleTriggerClick: () => setCurrentlyOpen(id),
+    }
+  }
+  triggerHandler.id = 0
```

Comment:
This requires a comment or two. What happens and is this necessary?

Reply:
@sebastiannicolajsen

Reply:
Should I add the comment in the source code? This is a generalized way of generating the two props `open` and `handleTriggerClick` Which allows us to implement custom behaviour on clicking the sidebar elements. Specifically, we want to only have one collapsible open at a time, which is why they all need a unique identifier which is selected as the one open (id). To ensure no conflict with other variables, we store the id on the function object.